Allow Running BuildRelease.ps1 Directly From Powershell / Normal Command Prompt

Right now, our `BuildRelease.ps1` script must be run from a Visual Studio command prompt:

```
powershell -ExecutionPolicy Bypass .\Nodejs\Setup\BuildRelease.ps1  -outdir c:\ntvs_out -dev -internal -vstarget "14.0" -skiptests -skipcopy -skipdebug
```

The build fails when run from a normal powershell session or normal command prompt.

```
Analysis\Analyzer\AnalysisUnit.cs(201,127): error CS1525: Invalid expression term '.' [C:\source\nodejstools\Nodejs\Product\Analysis\Analysis.csproj]
Analysis\Analyzer\AnalysisUnit.cs(201,128): error CS1003: Syntax error, ':' expected [C:\source\nodejstools\Nodejs\Product\Analysis\Analysis.csproj]
Analysis\Analyzer\AnalysisUnit.cs(201,176): error CS1525: Invalid expression term '.' [C:\source\nodejstools\Nodejs\Product\Analysis\Analysis.csproj]
Analysis\Analyzer\AnalysisUnit.cs(201,177): error CS1003: Syntax error, ':' expected [C:\source\nodejstools\Nodejs\Product\Analysis\Analysis.csproj]
Analysis\Analyzer\AnalysisUnit.cs(201,202): error CS1002: ; expected [C:\source\nodejstools\Nodejs\Product\Analysis\Analysis.csproj]
Analysis\Analyzer\AnalysisUnit.cs(201,202): error CS1525: Invalid expression term ')' [C:\source\nodejstools\Nodejs\Product\Analysis\Analysis.csproj]
Analysis\Analyzer\AnalysisUnit.cs(201,204): error CS1002: ; expected [C:\source\nodejstools\Nodejs\Product\Analysis\Analysis.csproj]
Analysis\Analyzer\AnalysisUnit.cs(234,33): error CS1525: Invalid expression term '.' [C:\source\nodejstools\Nodejs\Product\Analysis\Analysis.csproj]
Analysis\Analyzer\AnalysisUnit.cs(234,34): error CS1003: Syntax error, ':' expected [C:\source\nodejstools\Nodejs\Product\Analysis\Analysis.csproj]
  ReferenceGenerator -> C:\source\nodejstools\BuildOutput\Release14.0\ToolBinaries\ReferenceGenerator\NodeReferenceGenerator.exe
error CS1762 : Warning as error : A reference was created to embedded interop assembly 'c:\Program Files (x86)\Microsoft.NET\Primary Interop Assemblies\stdole.dll' because of an indirect re
ference to that assembly created by assembly 'c:\Program Files (x86)\Common Files\Microsoft Shared\MSEnv\PublicAssemblies\Microsoft.VisualStudio.CommandBars.dll'. Consider changing the 'Emb
ed Interop Types' property on either assembly. [C:\source\nodejstools\Nodejs\Product\InteractiveWindow\InteractiveWindow.csproj]
error CS1762 : Warning as error : A reference was created to embedded interop assembly 'c:\Program Files (x86)\Common Files\Microsoft Shared\MSEnv\PublicAssemblies\envdte.dll' because of an
 indirect reference to that assembly created by assembly 'c:\Program Files (x86)\Microsoft Visual Studio 14.0\VSSDK\VisualStudioIntegration\Common\Assemblies\v4.0\Microsoft.VisualStudio.She
ll.14.0.dll'. Consider changing the 'Embed Interop Types' property on either assembly. [C:\source\nodejstools\Nodejs\Product\InteractiveWindow\InteractiveWindow.csproj]
```

It'd be nice to be able to run this script directly from a git powershell command prompt.

--- a/Nodejs/Setup/BuildRelease.ps1
+++ b/Nodejs/Setup/BuildRelease.ps1
@@ -170,13 +170,13 @@ if ($release -or $mockrelease) {
 }
 
 # Get the path to msbuild for a configuration
-function msbuild-exe($target, $config) {
-	$toolpath = Get-ItemProperty -Path "HKLM:\Software\Wow6432Node\Microsoft\MSBuild\ToolsVersions\$($target.VSTarget)" -EA 0
-	$target_exe = $toolpath.MSBuildToolsPath + "msbuild.exe"
-	if (-not (Test-Path $target_exe)) {
-		Throw "Visual Studio build tools not found for $($target.VSTarget)."
-	}
-	$target_exe
+function msbuild-exe($target) {
+    $toolpath = Get-ItemProperty -Path "HKLM:\Software\Wow6432Node\Microsoft\MSBuild\ToolsVersions\$($target.VSTarget)" -EA 0
+    $target_exe = $toolpath.MSBuildToolsPath + "msbuild.exe"
+    if (-not (Test-Path $target_exe)) {
+        Throw "Visual Studio build tools $($target.VSTarget) not found."
+    }
+    $target_exe
 }
 
 # This function is used to get options for each configuration
@@ -472,7 +472,7 @@ try {
         
         foreach ($i in $target_info) {
             if (-not $skipbuild) {
-				$target_msbuild_exe = msbuild-exe $i
+                $target_msbuild_exe = msbuild-exe $i
                 $target_msbuild_options = msbuild-options $i
                 if (-not $skipclean) {
                     & $target_msbuild_exe /t:Clean $global_msbuild_options $target_msbuild_options $build_project
@@ -525,8 +525,8 @@ try {
                 }
                 submit_symbols "$project_name$spacename" "$buildnumber $($i.VSName) $config" "binaries" $i.signed_bindir $symbol_contacts
                 submit_symbols "$project_name$spacename" "$buildnumber $($i.VSName) $config" "symbols" $i.symboldir $symbol_contacts
-				
-				$target_msbuild_exe = msbuild-exe $i
+
+                $target_msbuild_exe = msbuild-exe $i
                 $target_msbuild_options = msbuild-options $i
                 & $target_msbuild_exe $global_msbuild_options $target_msbuild_options `
                     /fl /flp:logfile=$($i.signed_logfile) `

--- a/Nodejs/Setup/BuildRelease.ps1
+++ b/Nodejs/Setup/BuildRelease.ps1
@@ -169,6 +169,16 @@ if ($release -or $mockrelease) {
     $global_msbuild_options += "/p:ReleaseBuild=true"
 }
 
+# Get the path to msbuild for a configuration
+function msbuild-exe($target, $config) {
+	$toolpath = Get-ItemProperty -Path "HKLM:\Software\Wow6432Node\Microsoft\MSBuild\ToolsVersions\$($target.VSTarget)" -EA 0
+	$target_exe = $toolpath.MSBuildToolsPath + "msbuild.exe"
+	if (-not (Test-Path $target_exe)) {
+		Throw "Visual Studio build tools not found for $($target.VSTarget)."
+	}
+	$target_exe
+}
+
 # This function is used to get options for each configuration
 #
 # $target contains the following members:
@@ -190,7 +200,7 @@ if ($release -or $mockrelease) {
 #   signed_bindir       Output directory for signed binaries
 #   signed_msidir       Output directory for signed installers
 #   signed_unsigned_msidir  Output directory for unsigned installers containing signed binaries
-function msbuild-options($target, $config) {
+function msbuild-options($target) {
     @(
         "/p:VSTarget=$($target.VSTarget)",
         "/p:VisualStudioVersion=$($target.VSTarget)",
@@ -261,11 +271,6 @@ $supported_vs_versions = (
 #
 # #############################################################################
 # #############################################################################
-
-
-if (-not (Get-Command msbuild -EA 0)) {
-    Throw "Visual Studio build tools are required."
-}
 
 if (-not $outdir -and -not $release) {
     if (-not $outdir) {
@@ -467,11 +472,12 @@ try {
         
         foreach ($i in $target_info) {
             if (-not $skipbuild) {
+				$target_msbuild_exe = msbuild-exe $i
                 $target_msbuild_options = msbuild-options $i
                 if (-not $skipclean) {
-                    msbuild /t:Clean $global_msbuild_options $target_msbuild_options $build_project
+                    & $target_msbuild_exe /t:Clean $global_msbuild_options $target_msbuild_options $build_project
                 }
-                msbuild $global_msbuild_options $target_msbuild_options /fl /flp:logfile=$($i.logfile) $build_project
+                & $target_msbuild_exe $global_msbuild_options $target_msbuild_options /fl /flp:logfile=$($i.logfile) $build_project
 
                 if (-not $?) {
                     Write-Error "Build failed: $($i.VSName) $config"
@@ -519,9 +525,10 @@ try {
                 }
                 submit_symbols "$project_name$spacename" "$buildnumber $($i.VSName) $config" "binaries" $i.signed_bindir $symbol_contacts
                 submit_symbols "$project_name$spacename" "$buildnumber $($i.VSName) $config" "symbols" $i.symboldir $symbol_contacts
-
+				
+				$target_msbuild_exe = msbuild-exe $i
                 $target_msbuild_options = msbuild-options $i
-                msbuild $global_msbuild_options $target_msbuild_options `
+                & $target_msbuild_exe $global_msbuild_options $target_msbuild_options `
                     /fl /flp:logfile=$($i.signed_logfile) `
                     /p:SignedBinariesPath=$($i.signed_bindir) `
                     /p:RezipVSIXFiles=true `

--- a/Nodejs/Setup/BuildRelease.ps1
+++ b/Nodejs/Setup/BuildRelease.ps1
@@ -171,7 +171,7 @@ if ($release -or $mockrelease) {
 
 # Get the path to msbuild for a configuration
 function msbuild-exe($target) {
-    $msbuild_reg = Get-ItemProperty -Path "HKLM:\Software\Microsoft\MSBuild\ToolsVersions\$($target.VSTarget)" -EA 0
+    $msbuild_reg = Get-ItemProperty -Path "HKLM:\Software\Wow6432Node\Microsoft\MSBuild\ToolsVersions\$($target.VSTarget)" -EA 0
     if (-not $msbuild_reg) {
         Throw "Visual Studio build tools $($target.VSTarget) not found."
     }

--- a/Nodejs/Setup/BuildRelease.ps1
+++ b/Nodejs/Setup/BuildRelease.ps1
@@ -171,9 +171,13 @@ if ($release -or $mockrelease) {
 
 # Get the path to msbuild for a configuration
 function msbuild-exe($target) {
-    $toolpath = Get-ItemProperty -Path "HKLM:\Software\Wow6432Node\Microsoft\MSBuild\ToolsVersions\$($target.VSTarget)" -EA 0
-    $target_exe = $toolpath.MSBuildToolsPath + "msbuild.exe"
-    if (-not (Test-Path $target_exe)) {
+    $msbuild_reg = Get-ItemProperty -Path "HKLM:\Software\Microsoft\MSBuild\ToolsVersions\$($target.VSTarget)" -EA 0
+    if (-not $msbuild_reg) {
+        Throw "Visual Studio build tools $($target.VSTarget) not found."
+    }
+    
+    $target_exe = $msbuild_reg.MSBuildToolsPath + "msbuild.exe"
+    if (-not (Test-Path -Path $target_exe)) {
         Throw "Visual Studio build tools $($target.VSTarget) not found."
     }
     $target_exe

